src/Client.php

```diff
@@ -75,33 +79,29 @@ public function runQuery($query, bool $resultsAsArray = false, array $variables
      * @param string $queryString
      * @param bool   $resultsAsArray
      * @param array  $variables
+     * @param
      *
      * @return Results
      * @throws QueryError
      */
     public function runRawQuery(string $queryString, $resultsAsArray = false, array $variables = []): Results
     {
-        // Set request headers for authorization and content type
-        if (!empty($this->authorizationHeaders)) {
-            $options['headers'] = $this->authorizationHeaders;
-        }
+        $request = new Request('POST', $this->endpointUrl);
+        $headers = array_merge($this->authorizationHeaders, $this->httpHeaders, ['Content-Type' => 'application/json']);
 
-        // Set request options for \GuzzleHttp\Client
-        if (!empty($this->httpOptions)) {
-            $options = $this->httpOptions;
+        foreach($headers as $header => $value) {
+            $request = $request->withHeader($header, $value);
```

Comment:
My only concern about these changes is how you renamed the class variable `$httpOption` to `$httpHeaders` and how you're using it everywhere according to that understanding. In Guzzle these `httpOptions` could be things other than headers, such as certificates and timeout settings for example.
Check this pull request for instance:
https://github.com/mghoneimy/php-graphql-client/pull/14

I might be wrong on this, maybe every option in the Guzzle options can be converted to a header on the `GuzzleHttp\Psr7\Request`, I just want to make sure I understand how this works.

Reply:
At first, I removed the `httpOptions` property because:
- `Psr\Http\Client\ClientInterface::sendRequest` accepts only one argument, we can't pass the extra config options alongside the request
- the values that are being held in `httpOptions` can be passed into Guzzle's `Client` on instantiation, the way Guzzle applies these options should have the same result as if they were set on every request (at least from what I gathered form [Client:: configureDefaults](https://github.com/guzzle/guzzle/blob/master/src/Client.php#L219) and [Client:: prepareDefaults](https://github.com/guzzle/guzzle/blob/master/src/Client.php#L273))

I added it back because:
- [there is a check in the unit test suite](https://github.com/mghoneimy/php-graphql-client/blob/master/tests/ClientTest.php#L83) which started to fail after the changes described above. As it turned out, [Guzzle has a conditional logic](https://github.com/guzzle/guzzle/blob/ff7ed8feaa96350861ad6a28310fe365acff1aae/src/Client.php#L439) used for merging the "default" headers (set through the Client's constructor) with the headers set on the request manually. Only those of the "default" headers will be applied which are not set for the request object. This would have caused a backward compatibility issue (potentially).

Renamed `httpOptions` to `httpHeaders ` because:
- We don't need all the options, only the headers so we can merge them accordingly

Does this ☝️ make sense?

PS: While I was writing this I realized there might be a cleaner way of dealing with the problem + I didn't update the readme

Reply:
Yup, This makes perfect sense! :smile: 

I'm very happy too that the tests came in handy for catching this, I always have trouble testing things like request construction :smile: